addons/website_event_exhibitor/controllers/exhibitor.py

```diff
@@ -77,16 +77,26 @@ def _event_exhibitors_get_values(self, event, **searches):
         sponsor_types = sponsors_all.mapped('sponsor_type_id')
         sponsor_countries = sponsors_all.mapped('partner_id.country_id').sorted('name')
         # organize sponsors into categories to help display
-        sponsor_categories = dict()
+        sponsor_categories_dict = dict()
+        sponsor_categories = []
+        is_event_user = request.env.user.has_group('event.group_event_registration_desk')
         for sponsor in sponsors:
-            if not sponsor_categories.get(sponsor.sponsor_type_id):
-                sponsor_categories[sponsor.sponsor_type_id] = request.env['event.sponsor'].sudo()
-            sponsor_categories[sponsor.sponsor_type_id] |= sponsor
-        sponsor_categories = [
-            dict({
+            if not sponsor_categories_dict.get(sponsor.sponsor_type_id):
+                sponsor_categories_dict[sponsor.sponsor_type_id] = request.env['event.sponsor'].sudo()
+            sponsor_categories_dict[sponsor.sponsor_type_id] |= sponsor
+
+        for sponsor_category, sponsors in sponsor_categories_dict.items():
+            # To display random published sponsors first and random unpublished sponsors last
+            if is_event_user:
+                published_sponsors = sponsors.filtered(lambda s: s.is_published)
+                unpublished_sponsors = sponsors - published_sponsors
+                random_sponsors = sample(published_sponsors, len(published_sponsors)) + sample(unpublished_sponsors, len(unpublished_sponsors))
```

Comment:
Maybe use ``shuffle`` ?

Reply:
`shuffle` returns `None` (it shuffles the list itself in-place), so we'll have to store each recordset as a list of records in different variables(there are three recordsets here), shuffle them, and then use them. `sample` is simple :)